book/src/dyn_tutorial/permissions.md

```diff
@@ -0,0 +1,126 @@
+
+| Permission   | Name                               | Explanation                                                                                                                                                                                                                  |
+| ------------ | ---------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `my`         | Unique ownership                   | Nobody else has access to this object unless we grant it to them.                                                                                                                                                            |
+| `our`        | [Shared ownership](./share.md)     | Many variables can share access at once, but [none of them can write to the object](./sharing_xor_mutation.md).                                                                                                              |
+| `leased`     | [Unique lease](./lease.md)         | the object is [leased](./lease.md) from another variable. This variable has unique access, but it is only temporary. The owner can cancel the lease and start using the object again.                                        |
+| `our leased` | [Shared lease](./shared_leases.md) | a [shared lease](./shared_leases.md) means that multiple variables have ([read-only](./sharing_xor_mutation.md)) access, but that access is only temporary. The owner can cancel the lease and start using the object again. |
+| `any`        |                                    | the variable can store values with any of the above permisions                                                                                                                                                               |
+
+## Shared ownership
+
+For example, instead of writing `my p`, we could've written `our p` to declare a variable with shared ownership:
+
+```
+class Point(x, y)
+
+async fn main() {
+    our p = Point(22, 44)
+}
+```
+
+
+## Enforcing uniqueness and ownership
+
+When you declare a variable as `my p`, that means that it has *unique ownership* of the object(s) stored inside of it. 
+
+```
+async fn main() {
+    our p = Point(22, 44)
+    my q = p                                         # Error
+}
+```
+
+
+
+## Permissions and mutation
+
+The most obvious effect 
+## Default permissions
+
+If you don't specify any ownership permision, the default is `our leased`. This means that the variable cannot be used to mutate the fields of the object. Example:
+
+```
+class Point(x, y)
+
+async fn main() {
+    p = Point(22, 44)      # declares `p` in `our leased` mode
+    p.x += 1               # Error!
+}
+```
```

Comment:
I wonder why the default permission is `our leased`, not `our`.  
From my previous understanding of dada's ownership, the owner (unique or shared) has the responsibility for freeing the object, but the variable with `our leased` permission is not the owner of the object. So who is responsible for freeing the `Point(22, 44)`, is it `p`? 
```
class Point(x, y)
async fn main() {
    p = Point(22, 44)
}
```
Just thought it would make sense if this code was interpreted as 
```
class Point(x, y)
async fn main() {
    my tmp = Point(22, 44)
    p = tmp.lease.share
}
```
`Point(22, 44)` are freed once `tmp` has gone out of scope. I'm not sure if this is the correct way to understand.

Reply:
Good question! In Rust, terms, `our leased` is effectively `&T`, whereas `our` is effectively `Rc<T>`. I think the more common scenario is that functions take `&T` arguments (which they will use but not "capture") than `Rc<T>`.

To use a more dada-like answer, it is also affected by the subtyping and other rules (which I haven't described in the tutorial yet). The idea is that when you assign to a variable, if the variable demands owned content (`my`, `our`), you must give it something owned; if it demands unique access (`my`, `leased`), you must give it something unique. But if it demands leased, you can give it owned *or* leased, and similarly, you can assign a uniquely accessible thing into a shared location (it becomes shared). You can think of this as a kind of coercion:

* `my` coercible to `our`
* `my` coercible to `leased`
* `our` coercible to `our leased`
* `leased` coercible to `our leased`

By using `our leased` as the default, it means then that you can assign to that variable with virtually any value. This makes sense since "our leased" is basically saying "give me something I can read; I promise not to hold on to it", and that could be any of the permissions.

Reply:
@nikomatsakis 

> Good question! In Rust, terms, `our leased` is effectively `&T` [...]

shouldn't it be `&mut T`?

nice language btw, I'm excited about the future of dada!

Reply:
> shouldn't it be `Rc<&mut T>`?

Interesting question! I wasn't thinking of `our leased` as `our (leased T)` but more like `(our leased) T` -- in other words, it's like a lease that we collectively own. That said, I think we might want to make the concept of a "shared lease" more first class for other reasons, and I was thinking of using the keyword `shlease` for it (shared-lease, pronounced "shlease"), in which case this could be `shleased x = ...` I suppose. I'm not sure if I can tolerate that keyword but it's kind of amusing. 

I don't like the discrepancy between `our` and `shlease` though, nor do I love the discrepancy between `our` and `share`. It might be better to call `our` just `shared`. Then we'd have

* my
* shared
* leased
* shleased

`my` sort of stands out here, but `owned` just seems too long. =) And I did like the pithiness and playfullness of `our`!

Reply:
in my opinion, `owned` sounds better as a rust enthusiast.